Doc/library/asyncio-task.rst

```diff
@@ -390,7 +390,7 @@ Task
 .. function:: create_task(coro)
```

Comment:
I'm confused... where's the `name` argument? Please take another look at https://github.com/python/cpython/pull/8547#issuecomment-409245319.

Perhaps the problem is that we mean different things by "task factories":

* I mean custom user factories set by `loop.set_task_factory()` (which `loop.create_task()` later calls)
* You seem to mean `loop.create_task()`.

Maybe this is where the confusion is coming from.  Please re-read https://github.com/python/cpython/pull/8547#issuecomment-409245319.

Reply:
You seem to be suggesting that we ignore any `AttributeError` produced by a misbehaving task factory. This is something I've also considered and then rejected. Sweeping problems like that under the carpet is something I absolutely want to avoid. The best solution would, in my opinion, be to raise a user friendly error message explaining why setting a custom name for the task failed.

Reply:
> Sweeping problems like that under the carpet is something I absolutely want to avoid. The best solution would, in my opinion, be to raise a user friendly error message explaining why setting a custom name for the task failed.

Well, since task names are not critical in *any way* for asyncio programs I tend to disagree here. They are just here to simplify debugging.

I'm OK with issuing a warning though, like "some.lib.name.Task does not have a set_name() method".  We can later (in Python 3.9 or 4?) convert this into an error.

Reply:
A warning seems to be a reasonable compromise here. I will go add the `name` argument back.

Reply:
Great, thanks!

Reply:
For some reason I had missed parts of your commentary. It now makes a lot more sense to me. With the addition of the warning, I think we're finally on the same page. The one change I still want to make is to keep the `name` keyword argument in the stdlib Task constructor to avoid needlessly generating a task name if one was explicitly provided. Agreed?

Reply:
> For some reason I had missed parts of your commentary. It now makes a lot more sense to me. With the addition of the warning, I think we're finally on the same page.

Great, I had a feeling that something is off, but couldn't pinpoint that we mean different things when we discuss "task factories" :)

> The one change I still want to make is to keep the name keyword argument in the stdlib Task constructor to avoid needlessly generating a task name if one was explicitly provided.

OK by me.